src/QuadForm/Quad/ZLattices.jl

```diff
@@ -1611,3 +1612,205 @@ function overlattice(glue_map::TorQuadModMor)
+      Trem = Trem * tau1
+    else
+      bp = g + E[k,:]
+      qp = bp * G * transpose(bp)
+      @assert valuation(qp, p) <= gammaL[k] + 2*delta
+      tau1 = reflection(G, bp)
+      tau2 = reflection(G, E[k,:])
+      push!(reflection_vectors,bp)
+      push!(reflection_vectors,E[k,:])
+      Trem = Trem * tau1 * tau2
+    end
+    k += 1
+  end
+  reverse!(reflection_vectors)
+  R = reduce(*, reflection(G, v) for v in reflection_vectors)
+  err = valuation(T - R, p)
+  return err, reflection_vectors
+end
+
+
+function _isisometric_indef(L::ZLat, M::ZLat)
+  @req rank(L)>=3 "strong approximation needs rank at least 3"
+  @req degree(L)==rank(L) "lattice needs to be full for now"
+
+  # scale integral
+  n = rank(L)
+  s = scale(M)
+  M = rescale(M,s)
+  L = rescale(L,s)
+  @assert scale(M)==1
+  @assert scale(L)==1
+  g = genus(L)
+  if g != genus(M)
+    return false
+  end
+  if length(spinor_generators(g))==0
+    # unique spinor genus
+    return true
+  end
+  f, r = _isisometric_indef_approx(L, M)
+  return is_automorphous(g, r)
+end
+
+function _isisometric_indef_approx(L::ZLat, M::ZLat)
+  # move to same ambient space
+  qL = ambient_space(L)
+  diag, trafo = Hecke._gram_schmidt(gram_matrix(qL), identity)
+  qL1 = quadratic_space(QQ, diag)
+  L1 = lattice(qL1,basis_matrix(L)*inv(trafo))
+  @hassert :Lattice 1 genus(L1) == genus(L)
+  qM = ambient_space(M)
+  b,T = isisometric_with_isometry(qM, qL1)
+  @assert b  # same genus implies isomorphic space
+  M1 = lattice(qL1, T)
+  @hassert :Lattice 1 genus(M1) == genus(L)
+
+
+  V = ambient_space(L1)
+  gramV = gram_matrix(V)
+  sL = 8//scale(dual(L1))
+  bad = support(2*det(L1))
+  extra = 10
+  @label more_precision
+  targets = Tuple{fmpq_mat,fmpz,Int}[]
+  for p in bad
+    vp = valuation(sL, p) + 1
+    # precision seems to deteriorate along the number of reflections
+    precp = vp + 2*rank(L) + extra
+    # Approximate an isometry fp: Lp --> Mp
+    normalM1, TM1 = Hecke.padic_normal_form(gram_matrix(M1), p, prec=precp)
+    normalL1, TL1 = Hecke.padic_normal_form(gram_matrix(L1), p, prec=precp)
+    @assert normalM1 == normalL1
+    TT = inv(TL1) * TM1
+    fp = inv(basis_matrix(L1))* TT * basis_matrix(M1)
+    if valuation(det(fp)-1,p)<= vp
+      # we want fp in SO(Vp)
+      # compose with a reflection preserving Lp
+      norm_gen = _norm_generator(normalL1, p) * inv(TL1) * basis_matrix(L1)
+      @assert valuation((norm_gen * gramV * transpose(norm_gen))[1,1],p)==valuation(norm(L1), p)
+      fp = reflection(gramV, norm_gen) * fp
+      @assert valuation(det(fp)-1, p)>= vp
+    end
+    # double check that fp: Lp --> Mp
+    M1fp = lattice(V, basis_matrix(L1) * fp)
+    indexp = index(M1,intersect(M1fp, M1))
+    @assert valuation(indexp,p)==0
+    push!(targets,(fp, p, vp))
+  end
+  f = zero_matrix(QQ,0,0)
+  try
+    f = weak_approximation(V, targets)
+  catch e
+    if isa(e, InexactError)
+      extra = extra + 5
+      @goto more_precision
+    else
+      throw(e)
```

Comment:
```suggestion
      rethrow(e)
```
So that the stacktrace of the error thrown by `weak_approximation` is correct.